src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -0,0 +1,309 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading;
+
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// Activity represents operation with context to be used for logging.
+    /// Activity has operation name, Id, start time and duration, tags and baggage.
+    ///  
+    /// Current activity can be accessed with static AsyncLocal variable Activity.Current.
+    /// 
+    /// Activities should be created with constructor, configured as necessarily 
+    /// and then started with Activity.Start method which maintaines parent-child
+    /// relationships for the activities and sets Activity.Current.
+    /// 
+    /// When activity is finished, it should be stopped with static Activity.Stop method.
+    /// </summary>
+    public partial class Activity
+    {
+        /// <summary>
+        /// An operation name is a COARSEST name that is useful grouping/filtering. 
+        /// The name is typically a compile time constant.   Names of Rest APIs are 
+        /// reasonable, but arguments (e.g. specific accounts etc), should not be in
+        /// the name but rather in the tags.  
+        /// </summary>
+        public string OperationName { get; }
```

Comment:
In OpenTracing there are cases where you want to overwrite the operationName. E.g the hosting layer uses a generic "http-in" and the MVC layer overwrites it with "controller/action". 
Here this is problematic though because you use the operationName for the DiagnosticSource events as well. Not sure if/how to cover this?!